processing/src/main/java/org/apache/druid/query/groupby/strategy/GroupByStrategy.java

```diff
@@ -187,7 +188,10 @@ Sequence<ResultRow> processSubtotalsSpec(
    *
    * @return result sequence for the storage adapter
    */
-  Sequence<ResultRow> process(GroupByQuery query, StorageAdapter storageAdapter);
```

Comment:
I didn't do it in those methods since they take the subquery/subtotal result sequence in the method. I think the inlined subquery should execute using a separate query object and hence should emit their own metrics object. For pushed down subqueries, I'm not sure about their metrics framework as of now but my guess is that it should use the same metrics object as the outer one. will check and update